bundles/org.openhab.binding.nzwateralerts/src/main/java/org/openhab/binding/nzwateralerts/internal/handler/NZWaterAlertsHandlerFactory.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nzwateralerts.internal.handler;
+
+import static org.openhab.binding.nzwateralerts.internal.NZWaterAlertsBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link NZWaterAlertsHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.nzwateralerts", service = ThingHandlerFactory.class)
+public class NZWaterAlertsHandlerFactory extends BaseThingHandlerFactory {
+    @Nullable HttpClient httpClient = null;
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_WATERALERT);
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_WATERALERT.equals(thingTypeUID)) {
+            return new NZWaterAlertsHandler(thing, httpClient);
+        }
+
+        return null;
+    }
+
+    @Reference
+    protected void setHttpClientFactory(HttpClientFactory httpClientFactory) {
```

Comment:
I suggest populating the `httpClient` field in the `NZWaterAlertsHandlerFactory` constructor by just passing in `HttpClientFactory` as a `@Reference` parameter and adding `@Activate` above the constructor.

Doing this would allow you to make the `httpClient` field non-null and final and thus allow it to be non-null and final in the handler as well.

Reply:
Can you give me an example of how to do this please? I only work in Java for openHAB so not sure how to implement this.

Reply:
@cpmeister when you are able to. Cheers

Reply:
Got it sorted.